projects/sandbox/timeslide_injections/timeslide_injections.py

```diff
@@ -108,110 +191,196 @@ def main(
-
     else:
         # not considering segments so
         # make intersection from start to stop
         intersection = SegmentList([Segment(start, stop)])
 
-    # create list of timeslides
-    # for each ifo
-    timeslides = np.column_stack(
-        [
-            np.linspace(0, shift * (n_slides - 1), num=n_slides)
-            for shift in shifts
-        ]
+    total_length = sum([j - i for i, j in intersection])
+    logging.info(
+        "Querying {} segments of data totalling {} worth of data".format(
+            len(intersection), total_length
+        )
+    )
+
+    waveform_generator = get_waveform_generator(
+        waveform_approximant=waveform_approximant,
+        reference_frequency=reference_frequency,
+        minimum_frequency=fmin,
+        sampling_frequency=sample_rate,
+        duration=waveform_duration,
     )
+    if not prior_file.is_absolute():
+        prior_file = Path(__file__).resolve().parent / prior_file
+    priors = bilby.gw.prior.BBHPriorDict(str(prior_file))
 
-    for shifts in timeslides:
+    min_segment_length = min_segment_length or 0
+    total_slides = n_slides ** (len([i for i in shifts if i]))
+    max_shift = int(max(shifts) * n_slides * sample_rate)
 
-        # TODO: might be overly complex naming,
-        # but wanted to attempt to generalize to multi ifo
-        root = datadir / f"dt-{'-'.join(map(str,shifts))}"
+    process_pool = AsyncExecutor(4, thread=False)
+    thread_pool = AsyncExecutor(2, thread=True)
+    with process_pool, thread_pool:
+        for segment_start, segment_stop in intersection:
+            if segment_stop - segment_start < min_segment_length:
+                continue
 
-        # make root and timeslide directories
-        root.mkdir(exist_ok=True, parents=True)
+            # begin the download of data in a separate thread
+            logging.debug(
+                "Beginning download of segment {}-{}".format(
+                    segment_start, segment_stop
+                )
+            )
+            download_future = thread_pool.submit(
+                download_data,
+                ifos,
+                frame_type,
+                channel,
+                sample_rate,
+                segment_start,
+                segment_stop,
+            )
+            download_future.add_done_callback(
+                lambda f: logging.debug(
+                    "Completed download of segment {}-{}".format(
+                        segment_start, segment_stop
+                    )
+                )
+            )
 
-        # create TimeSlide object for injection
-        # this will create the directories if
-        # they don't exist
-        injection_ts = TimeSlide.create(root=root, field="injection")
+            # generate timestamps of trigger times for each waveform.
+            # Jitter will get added in `get_params`
+            signal_start = segment_start + buffer_
+            signal_stop = segment_stop + buffer_
```

Comment:
Should be `signal_stop = segment_stop - buffer_`

Reply:
Good catch, fixed